testcafe/tests/pages/photos-model.js

```diff
@@ -46,59 +47,159 @@ export default class Page {
     this.barPhotoBtnDelete = this.barPhoto.find('button').nth(2) //DELETE
     this.modalDelete = Selector('[class*="c-modal"]').find('div')
     this.modalDeleteBtnDelete = this.modalDelete.find('button').nth(2) //REMOVE
+
     this.allPhotos = Selector('[class*="pho-photo"]').find(
       '[class*="pho-photo-item"]'
     )
   }
 
+  async getPhotosCount(when) {
+    //@param {string} when : text for console.log
+    // wait until the 'exists' property of a selector is stable to count it
+    await wait(1000)
+      .until(this.allPhotos)
+      .exists.isStable()
+    const allPhotosCount = await this.allPhotos.count
+    console.log(
+      'Number of pictures on page (' + when + ' test):  ' + allPhotosCount
+    )
+    return allPhotosCount
+  }
+
   async uploadPhotos(files) {
     const numOfFiles = files.length
-    console.log('Uploading  ' + numOfFiles + ' picture(s)')
+    console.log('Uploading ' + numOfFiles + ' picture(s)')
 
     await t
       .setFilesToUpload(this.btnUpload, files)
       .expect(this.divUpload.visible)
       .ok('Upload pop-in does not show up')
       .expect(this.modalUpload.exists)
-      .ok({ timeout: 50000 })
+      .ok('Photo(s) not uploaded')
       .expect(this.divUpload.child('h4').innerText)
       .match(
         new RegExp('([' + numOfFiles + '].*){2}'),
         'Numbers of pictures uploaded does not match'
       )
     await t.takeScreenshot()
 
-    const allPhotosEndCount = await this.allPhotos.count //Pics count at the end
-    console.log(
-      'Number of pictures on page (After test):  ' + allPhotosEndCount
-    )
+    const allPhotosEndCount = await this.getPhotosCount('After')
     await t
       .expect(allPhotosEndCount)
       .eql(t.ctx.allPhotosStartCount + numOfFiles)
   }
 
-  async deletePhotos(numOfFiles) {
-    console.log('Deleting  ' + numOfFiles + ' picture(s)')
+  async selectPhotos(numOfFiles) {
+    console.log('Selecting ' + numOfFiles + ' picture(s)')
+
+    await t.hover(this.photoThumb(0)) //Only one 'hover' as all checkbox should be visible once the 1st checkbox is checked
 
-    await t.hover(page.photoThumb(0)) //Only one 'hover' as all checkbox should be visible once the 1st checkbox is checked
-    for (i = 0; i < numOfFiles - 1; i++) {
-      await t.click(page.photoCheckbox.nth(i)) //Index
+    var i
+    for (i = 0; i < numOfFiles; i++) {
```

Comment:
for(let i=0; i < numOfFiles; i++) 

C'est fini l'utilisation de `var` en JS ;)